File: app/src/main/java/br/com/jonathanzanella/myexpenses/expense/ExpenseInterface.java
package br.com.jonathanzanella.myexpenses.expense;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ExpenseInterface {
    @GET("expenses")
    Call<List<Expense>> index(@Query("last-updated-at") long lastUpdatedAt);
    @POST("expenses")
    Call<Expense> create(@Body Expense expense);
    @PUT("expenses/{id}")
    Call<Expense> update(@Path("id") String serverId, @Body Expense expense);
}

File: app/src/main/java/br/com/jonathanzanella/myexpenses/expense/ExpenseInterface.kt
package br.com.jonathanzanella.myexpenses.expense

import retrofit2.Call
import retrofit2.http.*

interface ExpenseInterface {
    @GET("expenses")
    fun index(@Query("last-updated-at") lastUpdatedAt: Long): Call<List<Expense>>

    @POST("expenses")
    fun create(@Body expense: Expense): Call<Expense>

    @PUT("expenses/{id}")
    fun update(@Path("id") serverId: String, @Body expense: Expense): Call<Expense>
}
